plugins/in_syslog/syslog_prot.c

```diff
@@ -31,88 +33,128 @@ static inline void consume_bytes(char *buf, int bytes, int length)
+                                         size_t message_size,
+                                         int message_type)
 {
-    int                i;
     int                result;
-    size_t             unpacker_offset;
-    msgpack_sbuffer    mp_sbuf;
-    msgpack_packer     mp_pck;
-    msgpack_unpacked   unpacked_buffer;
+    char              *modified_data_buffer;
+    int                modified_data_size;
+    msgpack_object_kv *new_map_entries[1];
+    msgpack_object_kv  message_entry;
     *result_buffer = NULL;
     *result_size = 0;
+    modified_data_buffer = NULL;
 
-    unpacker_offset = 0;
-    msgpack_unpacked_init(&unpacked_buffer);
-    result = msgpack_unpack_next(&unpacked_buffer,
-                                 base_object_buffer,
-                                 base_object_size,
-                                 &unpacker_offset);
+    if (message_key_name != NULL) {
+        new_map_entries[0] = &message_entry;
 
-    if (result != MSGPACK_UNPACK_SUCCESS) {
-        return -1;
-    }
+        message_entry.key.type = MSGPACK_OBJECT_STR;
+        message_entry.key.via.str.size = flb_sds_len(message_key_name);
+        message_entry.key.via.str.ptr  = message_key_name;
 
-    if (unpacker_offset != base_object_size) {
-        msgpack_unpacked_destroy(&unpacked_buffer);
-        return -2;
-    }
+        if (message_type == MSGPACK_OBJECT_BIN) {
+            message_entry.val.type = MSGPACK_OBJECT_BIN;
+            message_entry.val.via.bin.size = message_size;
+            message_entry.val.via.bin.ptr  = message_buffer;
+        }
+        else if (message_type == MSGPACK_OBJECT_STR) {
+            message_entry.val.type = MSGPACK_OBJECT_BIN;
+            message_entry.val.via.str.size = message_size;
+            message_entry.val.via.str.ptr  = message_buffer;
+        }
+        else {
+            result = FLB_MAP_NOT_MODIFIED;
 
-    if (unpacked_buffer.data.type != MSGPACK_OBJECT_MAP) {
-        msgpack_unpacked_destroy(&unpacked_buffer);
-        return -3;
-    }
+            return result;
+        }
 
-    msgpack_sbuffer_init(&mp_sbuf);
-    msgpack_packer_init(&mp_pck, &mp_sbuf, msgpack_sbuffer_write);
+        result = flb_msgpack_expand_map(base_object_buffer,
+                                        base_object_size,
+                                        new_map_entries, 1,
+                                        &modified_data_buffer,
+                                        &modified_data_size);
+        if (result == 0) {
+            result = FLB_MAP_EXPAND_SUCCESS;
+        }
+    }
 
-    msgpack_pack_map(&mp_pck, unpacked_buffer.data.via.map.size + 1);
+    if (result != FLB_MAP_EXPAND_SUCCESS) {
+        result = FLB_MAP_EXPANSION_ERROR;
 
-    for (i = 0; i < unpacked_buffer.data.via.map.size; i++) {
-        msgpack_pack_object(&mp_pck, unpacked_buffer.data.via.map.ptr[i].key);
-        msgpack_pack_object(&mp_pck, unpacked_buffer.data.via.map.ptr[i].val);
+        return result;
```

Comment:
Remove this and wrap lines 93-94 en an else to remove the innecessary premature return.

Reply:
Thanks for your review. I fixed it.